src/NUnitEngine/nunit.engine.core/Agents/RemoteTestAgent.cs

```diff
@@ -56,8 +57,9 @@ public class RemoteTestAgent : TestAgent, ITestEngineRunner
         /// Construct a RemoteTestAgent
         /// </summary>
         public RemoteTestAgent(Guid agentId, string agencyUrl, IServiceLocator services)
-            : base(agentId, services)
+            : base(services)
```

Comment:
If I understand you, you are saying that only Remote agents need an agent ID, as apposed to all agents. Of course, it makes no difference since we have not yet created any other agent types. Are you willing to give up the option of other agents?

Reply:
That's not quite it. All agents still need an ID and still use it to register with the agency. I moved the ID out of the ITestAgent interface so that ITestAgent members can represent remote procedure calls. I'm not planning for the protocol to have a "Start" message to the agent or a "Get ID" message. But even though its storage location moved, the ID is still being used in all the same ways as before this PR, I think.

Reply:
(And the reason to make ITestAgent look just like the wire protocol is to enable incremental changes to get there, rather than doing the whole thing in one PR and implementing our current ITestAgent interface on top of that.)